messaging/src/main/java/org/axonframework/eventhandling/gateway/DefaultEventGateway.java

```diff
@@ -16,100 +16,77 @@
-     * <p>
-     * The {@code dispatchInterceptors} are defaulted to an empty list.
-     * The {@link EventBus} is a <b>hard requirement</b> and as such should be provided.
+     * Creates a new {@link EventGateway} that uses the given {@code eventSink} to publish events. The
+     * {@code messageTypeResolver} is used to resolve the type of the event if no {@link EventMessage} is provided but a
+     * payload.
      *
-     * @return a Builder to be able to create a {@link DefaultEventGateway}
+     * @param eventSink           The {@link EventSink} to publish events to.
+     * @param messageTypeResolver The {@link MessageTypeResolver} to resolve the type of the event.
      */
-    public static Builder builder() {
-        return new Builder();
+    public DefaultEventGateway(@Nonnull EventSink eventSink, @Nonnull MessageTypeResolver messageTypeResolver) {
+        this.eventSink = Objects.requireNonNull(eventSink, "EventSink may not be null");
+        this.messageTypeResolver = Objects.requireNonNull(messageTypeResolver, "MessageTypeResolver may not be null");
     }
 
     @Override
     public void publish(@Nonnull List<?> events) {
-        this.publishAll(events);
+        AsyncUnitOfWork unitOfWork = new AsyncUnitOfWork();
+        unitOfWork.onInvocation(context -> {
+            publish(context, events);
+            return CompletableFuture.completedFuture(null);
+        });
+        unitOfWork.execute().join();
     }
 
-    /**
-     * A Builder class for {@link DefaultEventGateway}s.
-     * <p>
-     * The {@code dispatchInterceptors} are defaulted to an empty list.
-     * The {@link EventBus} is a <b>hard requirement</b> and as such should be provided.
-     */
-    public static class Builder extends AbstractEventGateway.Builder {
-
-        @Override
-        public Builder eventBus(@Nonnull EventBus eventBus) {
-            super.eventBus(eventBus);
-            return this;
-        }
-
-        @Override
-        public Builder dispatchInterceptors(
-                MessageDispatchInterceptor<? super EventMessage<?>>... dispatchInterceptors) {
-            super.dispatchInterceptors(dispatchInterceptors);
-            return this;
-        }
-
-        @Override
-        public Builder dispatchInterceptors(
-                List<MessageDispatchInterceptor<? super EventMessage<?>>> dispatchInterceptors) {
-            super.dispatchInterceptors(dispatchInterceptors);
-            return this;
-        }
-
-        @Override
-        public AbstractEventGateway.Builder messageNameResolver(MessageTypeResolver messageTypeResolver) {
-            assertNonNull(messageTypeResolver, "MessageNameResolver may not be null");
-            super.messageNameResolver(messageTypeResolver);
-            return this;
-        }
+    @Override
+    public void publish(@Nonnull ProcessingContext context, @Nonnull List<?> events) {
+        List<EventMessage<?>> convertedEvents = events.stream()
```

Comment:
```suggestion
        List<EventMessage<?>> eventMessages = events.stream()
```
Just a proposition. "Converted" suggest me that I changed something in the original one. But I just wrap it into message.